CMakeLists.txt

```diff
@@ -4,7 +4,12 @@ project(AtomicSharedPtr LANGUAGES CXX)
 
 cmake_minimum_required(VERSION 3.8)
 
-set(CMAKE_CXX_STANDARD 17)
+if (MSVC)
+    set(CMAKE_CXX_STANDARD 20)
+else()
+    set(CMAKE_CXX_STANDARD 17)
```

Comment:
I think global set(CMAKE_CXX_STANDARD 20) is fine. You use c++20 features later anyway (memory order namespace)